apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/DslJsonSerializer.java

```diff
@@ -524,10 +524,44 @@ private void serializeSpan(final Span span) {
         if (span.getContext().hasContent()) {
             serializeSpanContext(span.getContext());
         }
-        writeLastField("type", span.getType());
+        serializeSpanType(span);
         jw.writeByte(OBJECT_END);
     }
 
+    /**
+     * TODO: remove in 2.0
+     * To be removed for agents working only with APM server 7.0 or higher, where schema contains span.type, span.subtype and span.action
+     * @param span serialized span
+     */
+    private void serializeSpanType(Span span) {
+        writeFieldName("type");
+        String type = span.getType();
+        if (type != null) {
```

Comment:
If only the type is set, you can serialize it without checking for dots. You can then also remove the Utils.

```
if (type != null && subtype == null && action == null) {
    writeStringValue(type);
}
```

Reply:
That's what I thought of doing, but didn't seem right. Our model now includes these three fields for span and our entire code should assume that from now going forward. For example, we may decide to collect metrics for each DB type separately, or whatever. So an old API usage is supported, but I fit it into the model as soon as possible.

Reply:
Started with that as well 😄 (I mean- three setters), but it also didn't feel right, as it is only true now when I do this change. Otherwise, type, subtype and action are separate parts of the model, where you can set either elsewhere and in different times.
I got a good example for that when implemented the setting-through-tags for the OT bridge- each is done separately.
So I rather stick with that, but we can talk about it more. 
Anyway, I will add it as a static method to Span